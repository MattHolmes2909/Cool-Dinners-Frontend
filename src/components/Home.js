import React from "react";
import "../styles/Home.css";

const Home = () => {
  return (
    <div className="Home">
      <p className="welcome">Welcome to Cool Dinners!</p>
      <p className="login-message">Please login below:</p>
      <div className="login-form">
        <form>
          <label htmlFor="username" className="login-input">
            Username:{" "}
          </label>
          <input
            className="login-input input"
            type="text"
            placeholder="Your Username"
            name="username"
            required
          />
          <br />
          <label htmlFor="password" className="login-input">
            Password:{" "}
          </label>
          <input
            className="login-input input"
            type="password"
            placeholder="Your Password"
            name="password"
            required
          />
          <br />
          <button type="submit" className="login-form-button">
            Login
          </button>
          <br />
          <label>
            <input
              type="checkbox"
              checked="checked"
              name="remember"
              className="login-checkbox"
            />{" "}
            Remember me
          </label>
        </form>
      </div>
    </div>
  );
};

export default Home;
