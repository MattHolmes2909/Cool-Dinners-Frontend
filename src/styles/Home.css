
.Home {
  animation: foodMove 100s linear infinite;
  background-image: url("../images/backgroundfood.png");
  background-position: bottom;
  background-repeat: repeat-x;
  background-size: 10%;
  height: 85vh;
}

@keyframes foodMove {
  from {
    background-position-x: 0;
  }
  to {
    background-position-x: 100%;
  }
}

.welcome,
.login-message {
  margin: 0;
  text-align: center;
}

.welcome {
  font-size: 2rem;
}

.login-message {
  font-size: 1.5rem;
  margin-bottom: 1rem;
}

.login-form {
  animation: loginRev 1s forwards;
  background: conic-gradient(
    #dc3e43 20%,
    #e9740e 20%,
    #e9740e 40%,
    #d2e367 40%,
    #d2e367 60%,
    #4cbf5d 60%,
    #4cbf5d 80%,
    #5383e9 80%,
    #5383e9 100%
  );
  background-size: 300%;
  background-position: -500px;
  border-radius: 25px;
  line-height: 3rem;
  margin: auto;
  padding: 1rem;
  text-align: center;
  width: max-content;
}

.login-input {
  display: inline-block;
  width: 45%;
}

.input {
  border: 0;
  border-radius: 12px;
  font-size: 1rem;
  height: 1.5rem;
  text-align: center;
}

.input:focus {
  animation: inputGlow 0.5s forwards;
}

@keyframes inputGlow {
  to {
    background-color: #dce482;
  }
}

.login-form:focus-within {
  animation: login 1s forwards;
}

.login-checkbox {
  display: inline;
  width: 5%;
}

.login-form-button {
  background-color: #4cbf5d;
  border: 0;
  border-radius: 12px;
  font-family: "Bebas Neue", sans-serif;
  font-size: 1.5rem;
}

.login-form-button:hover {
  animation: loginGlowHover 0.5s forwards;
}

@keyframes login {
  from {
    background-position: -500px;
    background-size: 300%;
  }
  to {
    background-position: -200px;
    background-size: 200%;
  }
}

@keyframes loginRev {
  from {
    background-position: -200px;
    background-size: 200%;
  }
  to {
    background-position: -500px;
    background-size: 300%;
  }
}

@keyframes loginGlowHover {
  to {
    background-color: #15e673;
  }
}
