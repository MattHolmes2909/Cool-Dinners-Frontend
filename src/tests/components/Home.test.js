import React from "react";
import { render } from "@testing-library/react";
import Home from "../../components/Home";

describe("Home", () => {
    it("renders correctly", () => {
        const { asFragment } = render (<Home />);

        expect(asFragment()).toMatchSnapshot();
    });
});