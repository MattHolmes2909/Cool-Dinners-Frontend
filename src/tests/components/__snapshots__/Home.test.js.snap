// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<DocumentFragment>
  <div
    class="Home"
  >
    <p
      class="welcome"
    >
      Welcome to Cool Dinners!
    </p>
    <p
      class="login-message"
    >
      Please login below:
    </p>
    <div
      class="login-form"
    >
      <form>
        <label
          class="login-input"
          for="username"
        >
          Username: 
        </label>
        <input
          class="login-input input"
          name="username"
          placeholder="Your Username"
          required=""
          type="text"
        />
        <br />
        <label
          class="login-input"
          for="password"
        >
          Password: 
        </label>
        <input
          class="login-input input"
          name="password"
          placeholder="Your Password"
          required=""
          type="password"
        />
        <br />
        <button
          class="login-form-button"
          type="submit"
        >
          Login
        </button>
        <br />
        <label>
          <input
            checked=""
            class="login-checkbox"
            name="remember"
            type="checkbox"
          />
           Remember me
        </label>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
